src/main/java/sirius/biz/codelists/CustomLookupTable.java

```diff
@@ -90,22 +95,31 @@ protected Stream<LookupTableEntry> performSuggest(Limit limit, String searchTerm
         return Stream.concat(customTable.performSuggest(Limit.UNLIMITED, searchTerm, language),
                              baseTable.performSuggest(Limit.UNLIMITED, searchTerm, language))
                      .skip(limit.getItemsToSkip())
-                     .limit(limit.getMaxItems() == 0 ? Long.MAX_VALUE : limit.getMaxItems());
+                     .limit(limit.getMaxItems() == 0 ? Long.MAX_VALUE : limit.getMaxItems())
+                     .collect(filterCustomDuplicateCollector())
+                     .values()
```

Comment:
Why not `.filter` with a Predicate? Fells not performant to collect everything and build a new stream.

Reply:
filter can't access the other elements in the stream. Another solution would be something like:

```java
Map<String, LookupTableEntry> results = new LinkedHashMap<>();

customTable.performSuggest(Limit.UNLIMITED, searchTerm, language)
           .forEach(entry -> results.put(entry.getCode(), entry));

baseTable.performSuggest(Limit.UNLIMITED, searchTerm, language)
         .forEach(entry -> results.putIfAbsent(entry.getCode(), entry));

return results.values().stream().limit(limit.getMaxItems() == 0 ? Long.MAX_VALUE : limit.getMaxItems());
```

Implementing `LookupTableEntry#equals` which only compares the code and then using distinct here might be another solution.

In any case the limit needs to be the last part in the chain.

Reply:
I think you can try with a `.reduce()` As soon as you get 2 items with the same code, you gotta decide which one to keep.

Reply:
or maybe something like that could work as well. So you only need the codes in a temp set.

```java
Set<String> codes = new HashSet<>();
        return Stream.concat(customTable.performSuggest(Limit.UNLIMITED, searchTerm, language),
                             baseTable.performSuggest(Limit.UNLIMITED, searchTerm, language))
                     .skip(limit.getItemsToSkip())
                     .filter(codes::add)
                     .limit(limit.getMaxItems() == 0 ? Long.MAX_VALUE : limit.getMaxItems());
```